### PR TITLE
Updated atom ID representation order to match AtomGroup

### LINKED ISSUE #4181: repr of TopologyObjects should show order of underlying atom IDs
## Expected behavior ##
Representations of topology objects should list atom IDs in the order of the underlying AtomGroup.

## Actual behavior ##

I found it quite confusing when I looked at a Dihedral object and the order of atoms was different from my carefully crafted order to define the dihedral:
```python
>>> dih = u.atoms[[10, 0, 1, 3]].dihedral
>>> list(dih.atoms)
[<Atom 11: HG1 of type 3 of resname MET, resid 1 and segid 4AKE>,
 <Atom 1: N of type 56 of resname MET, resid 1 and segid 4AKE>,
 <Atom 2: HT1 of type 2 of resname MET, resid 1 and segid 4AKE>,
 <Atom 4: HT3 of type 2 of resname MET, resid 1 and segid 4AKE>]
>>> repr(dih)
'<Dihedral between: Atom 3, Atom 1, Atom 0, Atom 10>'
```

I’d much rather have the actual order output.

## Relevant code

Why do we do something special to the order of atom indices in an TopologyObject when printing the repr? Namely in https://github.com/MDAnalysis/mdanalysis/blob/4f36e2a06eb2d04274dfe4510910986c2dd17681/package/MDAnalysis/core/topologyobjects.py#L118 
```python
indices = (self.indices if self.indices[0] < self.indices[-1] else self.indices[::-1])
```
why do we reverse the order?

## Code to reproduce the behavior ##

``` python
>>> import MDAnalysis as mda
>>> from MDAnalysis.tests.datafiles import PSF, DCD,  GRO, PDB, TPR, XTC, TRR,  PRMncdf, NCDF
>>> u = mda.Universe(PSF, DCD)
>>> dih = u.atoms[[10, 0, 1, 3]].dihedral
>>> list(dih.atoms)
[<Atom 11: HG1 of type 3 of resname MET, resid 1 and segid 4AKE>,
 <Atom 1: N of type 56 of resname MET, resid 1 and segid 4AKE>,
 <Atom 2: HT1 of type 2 of resname MET, resid 1 and segid 4AKE>,
 <Atom 4: HT3 of type 2 of resname MET, resid 1 and segid 4AKE>]
>>> repr(dih)
'<Dihedral between: Atom 3, Atom 1, Atom 0, Atom 10>'
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 2.5.0
- Which version of Python (`python -V`)? 3.10.0
- Which operating system? macOS

### PR DESCRIPTION
**Pull request by ztimol**
Fixes #4181

Changes made in this Pull Request:
 - Removed check that reorders atom indices - confirmed that this is okay (see original issue).
 - Return topology object representation in order of underlying AtomGroup indices
 - Updated tests - topology object representation now matches order of AtomGroup indices

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [ ] Issue raised/referenced?

----
:books: Documentation preview :books:: https://mdanalysis--4191.org.readthedocs.build/en/4191/

### CODE REVIEW

File: package/MDAnalysis/core/topologyobjects.py
Comment (IAlibay) [inline package/MDAnalysis/core/topologyobjects.py]: Could you add a `versionchanged` in at the top of this class to indicate that this change has happened?

**Review by IAlibay — CHANGES_REQUESTED** [review]
Could you add an entry to changelog and I _think_ this is your first PR? (I can only see a closed in in the history), if so please also add yourself to AUTHORS.

File: package/MDAnalysis/core/topologyobjects.py
Comment (ztimol) [reply package/MDAnalysis/core/topologyobjects.py]: Updates made. Think they should be okay.

File: package/CHANGELOG
Comment (IAlibay) [inline package/CHANGELOG]: Could you:

1. Move this to the top of the list (we work by newest entry first in the changelog)
2. Keep this line to a maximum of 79 characters?

File: package/MDAnalysis/core/topologyobjects.py
Comment (IAlibay) [inline package/MDAnalysis/core/topologyobjects.py]:
```suggestion
        """Return representation in same order of AtomGroup indices"""
```
It's more common for docstring to use the triple " convention.

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple of minor changes and then I think we're good to go.

File: package/CHANGELOG
Comment (IAlibay) [inline package/CHANGELOG]: I forgot, can you also add your handle in the list of authors for 2.6.0 on line 16 please?

**Review by IAlibay — APPROVED** [review]
Sorry about the delay and thanks for the contribution!

I'll merge once CI returns green (had some codecov issues earlier today)

**Comment by ztimol** [comment]
Not sure if you've picked this up already but two tests seemed to have timed out. Can you reinitialise them when you get the chance. Thanks.

**Comment by ztimol** [comment]
Took a while but CI tests have finally all passed. Wondering if it was just an issue on the GH side of things. Would be great if this could be merged into develop. Thanks.

**Comment by IAlibay** [comment]
@ztimol the CI issues are related to #4209 I worked on it all weekend but unfortunately we've not really been able to work out what's up here.

Re; merge - unfortunately we are blocked by #4219, I'm hoping we'll merge it tonight and then we can start merging PRs again.

In the meantime, @ztimol could you please confirm that you agree to releasing this code under the terms of the [LGPLv2.1](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html) and that your contribution also adheres to the [developer certificate of origion](https://developercertificate.org/)?

**Review by IAlibay — CHANGES_REQUESTED** [review]
Blocking for license agreement

**Comment by ztimol** [comment]
No worries. Agree to [LGPLv2.1](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html) and can confirm adherence to [developer certificate of origion](https://developercertificate.org/).

**Review by IAlibay — APPROVED** [review]
All good, thanks for your patience @ztimol !

**Comment by IAlibay** [comment]
Just to clarify @ztimol this is LGPLv2.1+ (I typoed), i.e. you it may be relased under LGPLv2.1 or any later version.

**Pull request merged**
